* Patch from http://people.ubuntu.com/~fabbione/local-2.6.19-linux-libc-dev.diff

# DP: Description: Add missing bits in 2.6.19 kernel headers
# DP: Author: Jeff Bailey <jbailey@ubuntu.com>
# DP: Upstream status: Not Submitted.
# DP: Status Details: No response on http://sourceware.org/ml/libc-alpha/2006-10/msg00024.html
# DP: Date: 2006-10-27

diff -Naur glibc-2.5.orig/sysdeps/unix/sysv/linux/check_pf.c glibc-2.5/sysdeps/unix/sysv/linux/check_pf.c
--- glibc-2.5.orig/sysdeps/unix/sysv/linux/check_pf.c	2006-12-01 15:36:20.000000000 +0100
+++ glibc-2.5/sysdeps/unix/sysv/linux/check_pf.c	2006-12-01 15:36:31.000000000 +0100
@@ -30,10 +30,15 @@
 #include <asm/types.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#include <linux/if_addr.h>
 
 #include <not-cancel.h>
 #include <kernel-features.h>
 
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifinfomsg))
+#define IFA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifaddrmsg))))
+#define IFA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
 
 #ifndef IFA_F_TEMPORARY
 # define IFA_F_TEMPORARY IFA_F_SECONDARY
diff -Naur glibc-2.5.orig/sysdeps/unix/sysv/linux/ifaddrs.c glibc-2.5/sysdeps/unix/sysv/linux/ifaddrs.c
--- glibc-2.5.orig/sysdeps/unix/sysv/linux/ifaddrs.c	2006-12-01 15:36:20.000000000 +0100
+++ glibc-2.5/sysdeps/unix/sysv/linux/ifaddrs.c	2006-12-01 15:36:31.000000000 +0100
@@ -33,9 +33,14 @@
 #include <sysdep.h>
 #include <time.h>
 #include <unistd.h>
+#include <linux/if_addr.h>
 
 #include "netlinkaccess.h"
 
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifinfomsg))
+#define IFA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifaddrmsg))))
+#define IFA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
 
 /* We don't know if we have NETLINK support compiled in in our
    Kernel, so include the old implementation as fallback.  */
diff -Naur glibc-2.5.orig/sysdeps/unix/sysv/linux/if_index.c glibc-2.5/sysdeps/unix/sysv/linux/if_index.c
--- glibc-2.5.orig/sysdeps/unix/sysv/linux/if_index.c	2006-12-01 15:36:20.000000000 +0100
+++ glibc-2.5/sysdeps/unix/sysv/linux/if_index.c	2006-12-01 15:36:31.000000000 +0100
@@ -28,9 +28,14 @@
 #include <sys/ioctl.h>
 #include <bits/libc-lock.h>
 #include <not-cancel.h>
+#include <linux/if_addr.h>
 
 #include "netlinkaccess.h"
 
+#define IFA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifaddrmsg))))
+#define IFA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifinfomsg))
 
 /* Variable to signal whether SIOCGIFCONF is not available.  */
 # if __ASSUME_SIOCGIFNAME == 0
